### ISSUE TITLE
feat(cuda): enable TF32 Tensor Core math with graceful fallback for older GPUs

### ISSUE THREAD

**Issue by Samyssmile** [Samyssmile/edux#218]
## Problem

`CudaOps` does not configure cuDNN or cuBLAS math modes, leaving performance on the table on Ampere+ GPUs (RTX 3000+, A100, H100). TF32 uses Tensor Cores for float32 operations with identical training convergence but ~2–3x higher throughput.

**Critically, this must not break older GPUs** like Turing (GTX 1650 Ti, compute capability 7.5) which do not support TF32.

## Background

| GPU Generation | Compute Capability | TF32 Support |
|---|---|---|
| Turing (GTX 16xx, RTX 20xx) | 7.5 | ❌ No |
| Ampere (RTX 30xx, A100) | 8.x | ✅ Yes |
| Hopper (H100) | 9.0 | ✅ Yes |

When `CUDNN_TENSOR_OP_MATH` / `CUBLAS_TF32_TENSOR_OP_MATH` is set on a GPU that doesn't support TF32, cuDNN/cuBLAS simply **ignores the flag** and falls back to standard FP32 math. So the flag is safe to set unconditionally — but we should still detect and log the GPU capability for transparency.

### 1. Detect GPU Compute Capability at Init
```java
// During CudaOps initialization
int major = cudaDeviceGetAttribute(CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MAJOR, device);
int minor = cudaDeviceGetAttribute(CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MINOR, device);
boolean supportsTF32 = (major > 8) || (major == 8 && minor >= 0);
```

### 2. Set Math Mode
```java
// cuBLAS
cublasSetMathMode(cublasHandle, CUBLAS_TF32_TENSOR_OP_MATH);

// cuDNN — set on convolution descriptors
cudnnSetConvolutionMathType(convDesc, CUDNN_TENSOR_OP_MATH);
```

These calls are safe on all GPUs — pre-Ampere GPUs silently ignore them and use standard FP32.

### 3. Logging
On initialization, log the GPU name, compute capability, and whether TF32 is active:
```
[CUDA] NVIDIA GeForce RTX 3090 (compute 8.6) — TF32 Tensor Core math enabled
[CUDA] NVIDIA GeForce GTX 1650 Ti (compute 7.5) — TF32 not supported, using FP32
```

### 4. API Surface
- Enable TF32 by default (matches PyTorch default since 1.7)
- Provide opt-out: `NetworkBuilder.tf32(false)` for users who need bit-exact FP32 reproducibility
- No changes to the Tensor system, loss functions, or optimizer — TF32 is transparent at the cuBLAS/cuDNN level

## Compatibility Guarantee
- **Turing and older**: Code runs identically to today, no regression, no errors
- **Ampere and newer**: Automatic 2–3x speedup on matmul and convolution

## Expected Impact
**2–3x speedup** on Ampere+ GPUs for matmul and convolution operations — essentially free performance.

## References
- [NVIDIA TF32 Overview](https://blogs.nvidia.com/blog/tensorfloat-32-precision-format/)
- [cuBLAS Math Modes](https://docs.nvidia.com/cuda/cublas/#cublassetmathmode)
- [cuDNN Tensor Op Math](https://docs.nvidia.com/deeplearning/cudnn/developer-guide/)
- PyTorch: `torch.backends.cuda.matmul.allow_tf32` (default True since 1.7)
